markhor_tracks/src/markhor_hw_interface.cpp

```diff
@@ -109,7 +171,8 @@ void MarkhorHWInterface::write()
 
 void MarkhorHWInterface::read(const ros::Duration& period)
 {
-  // Read from the motor API, going to read from the TalonSRX objects
+    // Read from the motor API, going to read from the TalonSRX objects
+    //ROS_INFO("Vel: %d, %d", rear_right_drive->GetSensorCollection().GetQuadratureVelocity(),rear_left_drive->GetSensorCollection().GetQuadratureVelocity());
```

Comment:
Si ce n'est pas nécessaire comme code, ça serait best de l'enlever

Reply:
Ouais je l'enleverais pis j'implementerais une limite sur les drives direct avec forward/backward softlimitthreshold
On perdrais la differential logic here mais sa devrait etre géré par le diff controller pas le hw interface

Reply:
Ouin, je ne pense pas qu'on a beaucoup de communication avec le diff_drive_controller malheureusement. 

Aussi je ne suis pas certain de comprendre ce que tu veux dire par : 

> On perdrais la differential logic here mais sa devrait etre géré par le diff controller pas le hw interface

Genre quel differential logic qu'on perdrait ? C'est tu parce que les lirait ce que les drives output et on corrigerait en conséquence ?

Reply:
le diff_drive limite les outputs des drives techniquement par sa limite de linear et angular speed, si on le configure comme il faut, il ne vas jamais essayer de passer la limite de vitesse de nos drives, cependant en ce momment ce code la compense si le diff_drive envoie une commande trop grande, si on le mets sur les drives dirrect on perds ça mais à mon avis c un fix qui cache des issues plus deep de configuration, it should go mais si on l'enlève sa doit etre testé

Reply:
> cependant en ce momment ce code la compense si le diff_drive envoie une commande trop grande,

Attend, là on parle du code en commentaire qui est juste un `ROS_INFO`, est-ce que c'est vraiment de ça qu'on parle ? Si oui, il me manque un morceau.

Reply:
ahaha non, j'suis parti sur une tangente j'parlais de la func limitdifferentialspeed

Mais pendant qu'on est sur la fonction read, j'aimerais vraiment ça qu'on l'implémente correctement pour enable l'odometrie du diff_drive et respecter les standards du hw_interface

Reply:
Ouais certainement, cependant je te dirais est-ce que tu veux le faire dans cette PR ou bien tu voudrais créer une issue ? 

Aussi tu peux regarder comment dans cette répo il fait pour «parler» avec le controlleur, je ne suis pas certain que c'est comme cela, mais c'est sur ce code que je me suis basé pour faire l'implémentation des tracks. 
https://github.com/eborghi10/my_ROS_mobile_robot/blob/master/my_robot_base/include/my_robot_hw_interface.h#L81

Reply:
Aussi la fonction `limitdifferentialspeed` c'est une affaire que j'ai pris de du répo que je viens de t'envoyer, je suis même pas certain que c'est pertinent.

https://github.com/eborghi10/my_ROS_mobile_robot/blob/master/my_robot_base/include/my_robot_hw_interface.h#L157

On peut l'enlever sans problème.

Reply:
en fait dans un monde idéal y'aurait falu updater la lib CTRE et mettre les fix git windows, parce que live je peux pas commit dans le master a cause du bug, live j'aurais un hotfix a appliquer pour le launchscript gazebo mais je peux pas :'(

Reply:
> live j'aurais un hotfix a appliquer pour le launchscript gazebo mais je peux pas :'(

Tu ne peux pas juste faire une autre branche?